hi-ml/src/health_ml/utils/data_augmentations.py

```diff
@@ -171,17 +171,34 @@ def __init__(self, kernel_size: int, p: float = 0.5, min: float = 0.1, max: floa
         self.kernel_size = kernel_size
         self.p = p
 
-    def __call__(self, sample: torch.Tensor) -> torch.Tensor:
+    @staticmethod
+    def apply_gaussian_blur(sample: torch.Tensor, kernel_size: int, p: float, min: float, max: float) -> torch.Tensor:
         prob = np.random.random_sample()
-
-        if prob < self.p:
-            sigma = (self.max - self.min) * np.random.random_sample() + self.min
-            sample = np.array(sample.squeeze())  # type: ignore
-            sample = cv2.GaussianBlur(sample, (self.kernel_size, self.kernel_size), sigma)
-            sample = torch.Tensor(sample).unsqueeze(0)
-
+        if prob < p:
+            sigma = (max - min) * np.random.random_sample() + min
+            sample = sample.permute([0, 2, 3, 1]).squeeze().numpy()   # only 3 channels, color channel last
+            sample = cv2.GaussianBlur(sample, (kernel_size, kernel_size), sigma)
+            sample = torch.Tensor(sample).unsqueeze(0).permute(0, 3, 1, 2)
```

Comment:
This operations to go from channel first to channel last, and then back - are we doing that somewhere else too? If yes, this would be a good time to introduce 2 helper functions that can be tested in isolation and then re-used.

Reply:
Relatedly, do you remember if there's any reason we're using OpenCV for this and not `torchvision`'s [`gaussian_blur`](https://pytorch.org/vision/main/generated/torchvision.transforms.functional.gaussian_blur.html) (which operates on NCHW tensors)?

Reply:
I am not sure why we are not using it, but that would definitely simplify things. For our runs, the torchvision version takes much more time (~2x), perhaps this could be the reason?

Reply:
As discussed, will introduce the helper functions in a separate PR as it will affect both stain normalization and gaussian blur transforms